source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -747,26 +759,26 @@ DefaultCredentialsProviderChain::DefaultCredentialsProviderChain(
 
   if (!relative_uri.empty()) {
     const auto uri = absl::StrCat(CONTAINER_METADATA_HOST, relative_uri);
-    ENVOY_LOG(debug, "Using task role credentials provider with URI: {}", uri);
-    add(factories.createTaskRoleCredentialsProvider(api, context, fetch_metadata_using_curl,
-                                                    MetadataFetcher::create,
-                                                    CONTAINER_METADATA_CLUSTER, uri));
+    ENVOY_LOG(debug, "Using container role credentials provider with URI: {}", uri);
+    add(factories.createContainerRoleCredentialsProvider(api, context, fetch_metadata_using_curl,
+                                                         MetadataFetcher::create,
+                                                         CONTAINER_METADATA_CLUSTER, uri));
   } else if (!full_uri.empty()) {
-    const auto authorization_token =
+    auto authorization_token =
```

Comment:
This can still be const? Since AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE will be provided instead of  AWS_CONTAINER_AUTHORIZATION_TOKEN.

Meanwhile what is the reason not to read `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE` here? Would that file be dynamically updating?

Reply:
The reason behind removing const was because authorization_token provided in ENV variable (for ECS Task Role) is static, however the authorization_token provided in a file for EKS Pod Identity is regularly changing. I will change code path in refresh() instead and change this back to const.

Reply:
see below for new comment on this.